packages/popover/src/Popover.spec.tsx

```diff
@@ -104,4 +111,45 @@ describe('packages/popover', () => {
     unmount();
     expect(container.innerHTML).toBe('');
   });
+
+  test('fires `Transition` lifecycle hooks', async () => {
+    const callbacks = {
+      onEnter: jest.fn(),
+      onEntering: jest.fn(),
+      onEntered: jest.fn(),
+      onExit: jest.fn(),
+      onExiting: jest.fn(),
+      onExited: jest.fn(),
+    };
+    const { rerenderPopover } = renderPopover({
+      ...callbacks,
+    });
+
+    // Does not call any hooks on initial render
+    for (const cb of Object.values(callbacks)) {
+      expect(cb).not.toHaveBeenCalled();
+    }
+
+    // Calls enter callbacks when active is toggled to true
+    rerenderPopover({ active: true });
+
+    expect(callbacks.onEnter).toHaveBeenCalledTimes(1);
+    expect(callbacks.onEntering).toHaveBeenCalledTimes(1);
+    await waitFor(() => expect(callbacks.onEntered).toHaveBeenCalledTimes(1));
```

Comment:
ohhh we needed to await it

Reply:
Yeah, realized this might have been the issue you were seeing earlier